Services/Devices.cs

```diff
@@ -265,5 +372,14 @@ private Azure.Devices.Client.DeviceClient GetDeviceSdkClient(Device device, IoTH
 
             return sdkClient;
         }
+
+        private IEnumerable<IEnumerable<T>> SplitArray<T>(IReadOnlyCollection<T> array, int size)
+        {
+            var count = (int) Math.Ceiling((float) array.Count / size);
```

Comment:
Unfortunate that there's no built in array function that allows splitting of arrays into sub-arrays.  I looked around & couldn't find a better way to do this.  did you consider linq?  something like:

arrayString.GroupBy(s => i++/lengthToSplit).Select(g => g.ToArray()).ToList()

Reply:
nice, I like linq. I find that "group by" odd to read though. Future-wise we'll have to do a lot of partitioning, e.g. to run across multiple instances, so I'm sure we'll revisit this code